### PR TITLE
[marvell-arm64]: Add support for 98DX35xx and 98CX85xx platform

### PR DESCRIPTION
**Pull request by pavannaregundi**
These patches add necessary support for marvell platforms - 98DX35xx and 98CX85xx.

Testing:
Build of target/sonic-marvell-arm64.bin, load of the image on AC5XRD(98DX35xx ) and Falcon(98CX85xx) switches, verified that all dockers operational and “show interface status” displays all ports correctly.

Build of target/sonic-marvell-armhf.bin, load of the image on M0, verified that all dockers operational and “show interface status” displays all ports correctly.

### CODE REVIEW

**Comment by paulmenzel** [comment]
Please write more [elaborate commit messages](https://cbea.ms/git-commit/) (for your commit and the added patches). SONiC requires the added patches to be sent upstream, so they do not need to be maintained by the SONiC community longterm.

**Comment by pavannaregundi** [comment]
@paulmenzel Sub commits do have description. Please check them.

**Comment by paulmenzel** [comment]
Yes, they are much better than before. But they do not follow common patterns (statements (with a verb in imperative mood), no dot/period at the end of commit message summaries). Also, the Gerrit change-Ids are not needed, unless you tell us the public Gerrit instance. The missing commit message more referred to the added patch files.

On what device did you test this?

**Comment by pavannaregundi** [comment]
@paulmenzel I have corrected some commit messages. Some part of the patches are upstreamed, those details are mentioned along with commit message. 
Testing details are mentioned above.

**Comment by radha-danda** [comment]
@paulmenzel, can you please review the PR? All your comments are addressed by @pavannaregundi

**Comment by carl-nokia** [comment]
@paulmenzel,  we are struggling a bit to establish a new arm64 pipeline for Microsoft Azure CI/CD and this PR with arm64 support in addition to armhf is the initial foundation for several other PR's to establish the new arm64 pipeline in addition to the existing armhf pipeline.   -- your attention is greatly appreciated; are there any issues we could assist with?

**Comment by paulmenzel** [comment]
Sorry, the commit messages (of the commits in the merge/pull request) still do not follow the common standard. Additionally, the added patch files also are not properly created with `git format-patch`. Please look at the merge/pull requests from Nvidia how to do it.

**Comment by carl-nokia** [comment]
Thanks Paul -- we will set up a team review of this PR and address your comments by Monday.

**Comment by pavannaregundi** [comment]
> Sorry, the commit messages (of the commits in the merge/pull request) still do not follow the common standard. Additionally, the added patch files also are not properly created with `git format-patch`. Please look at the merge/pull requests from Nvidia how to do it.

@paulmenzel I have pushed changes for 'git format-patch' and also addressed commit message to follow common standards. Please see if they are fine now and review the PR.

**Comment by pavannaregundi** [comment]
@Pavan-Nokia  please review PR

**Comment by carl-nokia** [comment]
@paulmenzel -- we have both companies meeting tomorrow to review anything needed for this PR  again at 9:30AM EST -- any chance you could join?     Thanks

File: patch/0003-mvneta-setup-xbar-windows-for-ac5.patch [hunk 1]
```diff
@@ -0,0 +1,66 @@
+From: Pavan Naregundi <pnaregundi@marvell.com>
+Date: Mon, 20 Feb 2023 04:11:01 +0000
+Subject: mvneta: setup xbar windows for ac5
```
Comment (paulmenzel) [inline patch/0003-mvneta-setup-xbar-windows-for-ac5.patch:3]: The verb is spelled with a space: Set up …

File: patch/0003-mvneta-setup-xbar-windows-for-ac5.patch [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
++	for (i = 0; i < 6; i++) {
++		mvreg_write(pp, MVNETA_WIN_BASE(i), 0);
++		mvreg_write(pp, MVNETA_WIN_SIZE(i), 0);
++
++		if (i < 4)
++			mvreg_write(pp, MVNETA_WIN_REMAP(i), 0);
++	}
++
++	/*
++	 * Setup window #0 base 0x0 to target XBAR port 2 (AMB2), attribute 0xb,
```
Comment (paulmenzel) [inline patch/0003-mvneta-setup-xbar-windows-for-ac5.patch:43]: Ditto.

File: patch/0003-mvneta-setup-xbar-windows-for-ac5.patch [hunk 3]
```diff
@@ -0,0 +1,66 @@
+From: Pavan Naregundi <pnaregundi@marvell.com>
+Date: Mon, 20 Feb 2023 04:11:01 +0000
+Subject: mvneta: setup xbar windows for ac5
+
```
Comment (paulmenzel) [inline patch/0003-mvneta-setup-xbar-windows-for-ac5.patch:4]: Commit message body is missing.

File: patch/0004-support-be-for-ac5-spi-driver.patch [hunk 4]
```diff
@@ -0,0 +1,36 @@
+From: Pavan Naregundi <pnaregundi@marvell.com>
+Date: Mon, 20 Feb 2023 04:11:02 +0000
+Subject: support be for ac5 spi driver
```
Comment (paulmenzel) [inline patch/0004-support-be-for-ac5-spi-driver.patch:3]: What is “be”?

File: patch/0006-mv6xxx-i2c-bus-lock-arb-loss.patch [hunk 5]
```diff
@@ -0,0 +1,217 @@
+From: Pavan Naregundi <pnaregundi@marvell.com>
+Date: Mon, 20 Feb 2023 04:11:06 +0000
+Subject: mv6xxx: i2c bus lock (arb-loss)
```
Comment (paulmenzel) [inline patch/0006-mv6xxx-i2c-bus-lock-arb-loss.patch:3]: No statement.

File: patch/0006-mv6xxx-i2c-bus-lock-arb-loss.patch [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+
+Some i2c slaves, mainly SFPs, might cause the bus to
```
Comment (paulmenzel) [inline patch/0006-mv6xxx-i2c-bus-lock-arb-loss.patch:5]: Please reflow for 75 characters per line.

File: patch/0001-dts-support-for-7020-comexpress.patch [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+index 293403a1a..c27ce27b9 100644
+--- a/arch/arm64/boot/dts/marvell/armada-70x0.dtsi
++++ b/arch/arm64/boot/dts/marvell/armada-70x0.dtsi
+@@ -47,6 +47,12 @@ &cp0_syscon0 {
+ 	cp0_pinctrl: pinctrl {
+ 		compatible = "marvell,armada-7k-pinctrl";
+ 
++		sdhci_pins: sdhi-pins {
++			marvell,pins = "mpp56", "mpp57", "mpp58",
++							"mpp59", "mpp60", "mpp61", "mpp62";
```
Comment (paulmenzel) [inline patch/0001-dts-support-for-7020-comexpress.patch:175]: Indentation looks incorrect.

File: patch/0001-dts-support-for-7020-comexpress.patch [hunk 8]
```diff
@@ -0,0 +1,184 @@
+From: Pavan Naregundi <pnaregundi@marvell.com>
+Date: Mon, 20 Feb 2023 04:10:57 +0000
+Subject: dts support for 7020 comexpress
```
Comment (paulmenzel) [inline patch/0001-dts-support-for-7020-comexpress.patch:3]: No statement.

File: patch/0001-dts-support-for-7020-comexpress.patch [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+
```
Comment (paulmenzel) [inline patch/0001-dts-support-for-7020-comexpress.patch:4]: Missing commit message body.

**Review by paulmenzel — CHANGES_REQUESTED** [review]
I looked at it ten minutes and already found several cases, that should be improved. No idea, why you do not see it when doing your team meetings.

It’s also not clear, what patches were sent upstream for review (the maintainers are able to “judge” the correctness better than the SONiC kernel folks), and what patches are included already upstream.

For me, I am doing this in my free time, so this was the last comment on this merge/pull request. Especially as I had a bad experience with a Nokia contribution in the past, where promises were made and broken. The other project members should comment and review from now on.

**Comment by carl-nokia** [comment]
@paulmenzel Thanks for your help Paul.  Review focus on this Marvell generated PR had been on technical and operational aspects ensuring a new arm64 target could be built and run without any impact on existing armhf capabilities.  Marvell is working on addressing your feedback for correction. Thank You

**Comment by pavannaregundi** [comment]
Thanks @paulmenzel for review comments. We will raise a new PR after addressing review comments.

**Pull request closed without merging**